### PR TITLE
chore: remove version requirement

### PR DESCRIPTION
**Pull request by pgautier404**
This commit removes the minimum language version directive in our go.mod and go.work files and updates the `CONTRUBITING` doc to specify a mimimum version of 1.18 as detailed in #190.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ any of the operational overhead required by traditional caching solutions!
 
 ### Requirements
 
-- [Go version 1.19](https://go.dev/dl/) or newer.
```
Comment (schwern) [inline README.md:21]: This should go into the README template.

**Review by schwern — COMMENTED** [review]
Is there a reason we are not documenting the minimum in the README?

CONTRIBUTING says 1.18, but go.mod says 1.16. Is this correct?

File: examples/go.mod [hunk 2]
```diff
@@ -1,19 +1,8 @@
 module github.com/momentohq/go-example
 
-go 1.19
+go 1.16
```
Comment (poppoerika) [inline examples/go.mod:3]: If the minimum version we chose is `1.18`, that should be reflected here as well? or am I missing something?

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/momentohq/client-sdk-go
 
-go 1.19
+go 1.16
```
Comment (poppoerika) [inline go.mod:3]: Same question as [here](https://github.com/momentohq/client-sdk-go/pull/234/files#r1127169755).

File: go.work [hunk 4]
```diff
@@ -1,5 +1,3 @@
-go 1.19
-
```
Comment (poppoerika) [inline go.work:2]: Is there a specific reason why we want to remove the Go version the files should be interpreted with here?

File: examples/go.mod (hunk 2 above)
Comment (pgautier404) [reply examples/go.mod:3]: Sorry, this actually isn't ready for review. Feel free to take a look at https://github.com/momentohq/client-sdk-go/issues/190 to see where I am with this. Didn't mean for anyone to start reviewing this until I can get those issues resolved.

Comment (poppoerika) [reply examples/go.mod:3]: No worries! Roger that.

**Comment by pgautier404** [comment]
Closing. Will restart in a separate PR.

**Pull request closed without merging**